pkg/backend/httpstate/client/marshal.go

```diff
@@ -36,13 +36,9 @@ var jsonIterConfig = jsoniter.Config{SortMapKeys: true}.Froze()
 // Injects newlines to allow efficient textual diffs over the JSON. Textual diffs currently use O(N^2) memory in the
 // number of newlines, so the injection needs to be conservative. Currently it limits to up to maxNewLines newlines
 // which would result in max 8MB memory use by the algorithm.
-func MarshalUntypedDeployment(deployment *apitype.DeploymentV3) (json.RawMessage, error) {
-	var buf bytes.Buffer
+func MarshalUntypedDeployment(b *bytes.Buffer, deployment *apitype.DeploymentV3) error {
```

Comment:
Functional ..options pattern here if want to avoid making this a breaking change, though perhaps under pkg/ we do not care.

Reply:
Ah never mind, return type is changing too. Then it could be a new func.

Reply:
We're generally fine with breaking changes in 'pkg' apart from codegen/plugin bits still used by providers.